src/main/java/net/minecraftforge/items/SlotItemHandler.java

```diff
@@ -126,7 +126,8 @@ public int getItemStackLimit(@Nonnull ItemStack stack)
     @Override
     public boolean canTakeStack(EntityPlayer playerIn)
     {
-        return !this.getItemHandler().extractItem(index, 1, true).isEmpty();
+        net.minecraft.util.EnumActionResult cancelResult = net.minecraftforge.common.ForgeHooks.onItemTake(playerIn, this.getStack());
```

Comment:
I added the event triggering code to `SlotItemHandler#canTakeStack()` if I understood @liach and @raoulvdberge correctly, this should do the trick.